### PR TITLE
Bug 1190260 - Convert action menu remember checkbox to gaia-checkbox

### PR DESCRIPTION
**Pull request by KevinGrandon**
https://bugzilla.mozilla.org/show_bug.cgi?id=1190260

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3fb6eec67427d2c1b72b583fb2a078ef32f8fe7b

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1a6bb282fc9287c5035cb9b6aed16deb232b54a5

File: apps/system/js/action_menu.js [hunk 1]
```diff
@@ -107,7 +107,7 @@
     var target = evt.target;
     var defaultSelected = false;
     if (this.askForDefaultChoice) {
-      defaultSelected = !!this.defaultChoiceInput.getAttribute('checked');
+      defaultSelected = !!this.defaultChoiceInput.checked;
```
Comment (timdream) [inline apps/system/js/action_menu.js]: Drop the `!!`; `checked` is always a boolean.

Comment (timdream) [inline apps/system/js/action_menu.js]: Oh wait, `defaultChoiceInput` is a `<gaia-checkbox>`, not a native DOM element. You would need to make sure `<gaia-checkbox>` impl do that first.

File: shared/elements/gaia_checkbox/style.css [hunk 2]
```diff
@@ -13,6 +13,7 @@ gaia-checkbox {
 
 #checkbox span {
   font-size: 1.9rem;
+  font-weight: normal;
```
Comment (timdream) [inline shared/elements/gaia_checkbox/style.css]: I thought we need to upstream changes like this?

File: apps/system/js/action_menu.js [hunk 3]
```diff
@@ -120,23 +120,18 @@
   };
 
   ActionMenu.prototype._appendDefault = function() {
-    var defaultChoice = document.createElement('label');
-    defaultChoice.setAttribute('class', 'pack-checkbox');
+    var defaultChoice = document.createElement('gaia-checkbox');
+    defaultChoice.className = 'inline';
+    defaultChoice.configureClass();
```
Comment (timdream) [inline apps/system/js/action_menu.js]: Really? This API design is questionable. But it's not related to this patch though.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=edb81aa0076da016dad9953a54ff350cbd9fa1f2

File: apps/system/js/action_menu.js (hunk 1 above)
Comment (KevinGrandon) [inline apps/system/js/action_menu.js]: It does and I'll drop it, was just preserving the previous syntax.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a443cce1d340ead9bbf56b3445e23c52635b6c14

File: apps/system/js/action_menu.js (hunk 3 above)
Comment (KevinGrandon) [inline apps/system/js/action_menu.js]: I'll make a note and follow this up with a fix to the components.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ead127306fe982902f3bcf8dddfda317f1729d28

File: shared/elements/gaia_checkbox/style.css (hunk 2 above)
Comment (KevinGrandon) [inline shared/elements/gaia_checkbox/style.css]: This *is* upstream for 2.5. The components in shared/ are different than the components in the gaia-components repo.

**Pull request merged**